### PR TITLE
Add CBech32Data

### PR DESCRIPTION
**Pull request by inaltoasinistra**
The CBech32Data class implements an interface similar to CBase58Data, it encapsulates the python [reference implementation](https://github.com/sipa/bech32/tree/master/ref/python) of bech32 addresses.

### CODE REVIEW

**Comment by petertodd** [comment]
Looks like a good start!

We should think how we can present a decent API for this, along the lines of the existing `CBitcoinAddress` scheme. I haven't had a chance to review what Bitcoin Core is doing for that yet, and of course, that pull-req isn't quite merged yet either, so I think it's fine to wait until that happens and we're sure BIP173 is stabilized.

File: bitcoin/__init__.py [hunk 1]
```diff
@@ -52,6 +54,7 @@ class RegTestParams(bitcoin.core.CoreRegTestParams):
     BASE58_PREFIXES = {'PUBKEY_ADDR':111,
                        'SCRIPT_ADDR':196,
                        'SECRET_KEY' :239}
+    BECH32_HRP = 'tb'
```
Comment (maltemoeser) [inline bitcoin/__init__.py]: [Should](https://github.com/satoshilabs/slips/blob/master/slip-0173.md) be `bcrt` for regtest

**Comment by dbolser** [comment]
Yo @petertodd, can you make @dgpv or @Simplexum co-maintainer of this library?

**Comment by dgpv** [comment]
Not applicable, as explained in https://github.com/petertodd/python-bitcoinlib/issues/192#issuecomment-430898352 - I am not ready to maintain original library, only our derivative library, that have reduced scope.

**Comment by dgpv** [comment]
This was merged via #189

**Pull request closed without merging**